### ISSUE TITLE
Compilation fails if storyboard doesn't contain initial VC

### ISSUE THREAD

**Issue by vkaramov** [AliSoftware/SwiftGen#160]
See generated code below:

```
struct StoryboardScene {
  enum ActivateBaseTariff: String, StoryboardSceneType {
    static let storyboardName = "ActivateBaseTariff"

    static func initialViewController() ->  {
      guard let vc = storyboard().instantiateInitialViewController() as?  else {
        fatalError("Failed to instantiate initialViewController for \(self.storyboardName)")
      }
      return vc
    }
```

**Comment by AliSoftware** [comment]
Hi vkaramov,

What version of SwiftGen are you using, and which Storyboards template are you using exactly?

**Comment by vkaramov** [comment]
I prefer storyboards-lowercase.stencil and I'm using version 2.0.0

**Comment by AliSoftware** [comment]
`storyboards-lowercase.stencil` from version 2.0.0 doesn't include such generated code.

I'm guessing that you're instead using the `master` branch and not the 2.0.0 tag, because such generated code comes from #153, which was merged in master but not released yet.

If so, maybe add a comment/reference in that PR to let @ikesyo know.
To fix this, there's probably just a unit test to add for that use case + a conditional in the templates in case `{{ storyboard.initialScene }}` is empty. Should be an easy PR to do.

**Comment by vkaramov** [comment]
Yes, I'm using 2.0, but took template from master, cause I migrated from 0.7.4 and wanted to keep things under control. Solved for me, by removing initial VC from the template

**Comment by AliSoftware** [comment]
Ok thx for the feedback. We'll still need to fix the bundled template on master before the next release to address this use case!

**Comment by ikesyo** [comment]
My answer is here: https://github.com/AliSoftware/SwiftGen/pull/153#issuecomment-242974233.

**Comment by ikesyo** [comment]
> to address this use case!

I'm sot sure if that is really an expected use case 😕

**Comment by AliSoftware** [comment]
> I'm sot sure if that is really an expected use case 😕

I never personally related Storyboards with NO initial scene, but I believe some might want that to work. Technically speaking, it's still valid to create Storyboards with multiple scenes in it but none being the initial one.

For example, i've already seen in a coworker's project a `PaymentMethods.storyboard` which had various scenes (for `CreditCardViewController`, `PaypalViewController`, `ApplePayViewController` and whatnot), and none of those scene would make sense to be the initial one, as the one scene to create would depending on the payment method the user chose at the previous step.
In such use case, that would make sense to create that `PaymentMethods.storyboard` with all scenes having sceneIdentifiers and none of them being the initial one, so I believe that's still a use case we should handle, especially if it's one `{% if storyboard.initialScene %}` test away.

**Comment by ikesyo** [comment]
There should be a misunderstanding between me and you. I thought that `this use case` what you said above means [`using a new template with an old version`](https://github.com/AliSoftware/SwiftGen/pull/153#issuecomment-242974809).

> I never personally related Storyboards with NO initial scene, but I believe some might want that to work. Technically speaking, it's still valid to create Storyboards with multiple scenes in it but none being the initial one.

The use case of a storyboard which does not have `initialViewController` is tested and should work on current `master` as I answered in https://github.com/AliSoftware/SwiftGen/pull/153#issuecomment-242977055.

**Comment by AliSoftware** [comment]
Oh right, sorry. By "that use case" I totally meant "a storyboard with no initial scene". We can't imagine supporting new templates with older SwiftGen versions, that wouldn't make sense to try to maintain such complex retro-compatibility anyway and is out of the question. You're totally not supposed to use templates from the a new major version with an older major version.

**Comment by AliSoftware** [comment]
Fixed via #163
